src/compute/src/render/top_k.rs

```diff
@@ -367,7 +367,7 @@ where
             G: Scope,
             G::Timestamp: Lattice,
         {
-            let mut aggregates = HashMap::new();
+            let mut aggregates = BTreeMap::new();
```

Comment:
I'm slightly concerned about this one, since it seems to be running in a hot loop. So maybe the perf difference is important.

Reply:
I'm running the feature benchmarks to see if there is a regression: https://buildkite.com/materialize/nightlies/builds/1723#0185f3a1-8d2d-480c-ac72-be790909654c. Apart from that, do we have any other benchmarks that are a good test for TopK?

Reply:
@vmarcos did some work on determining the performance of various changes related to top-k. I agree that changing the `HashMap` to `BTreeMap` might have a performance implication here, especially for large key domains, which happen if the group key has many variants. The order in which we produce outputs is not something downstream operators should rely on, which means that there isn't any semantic difference between the two implementations here.

Reply:
@vmarcos Are the tests you did documented? And do you think they would be sufficient to inform if we should switch the map implementation or not?

We can either run the benchmarks and use them to decide, or not run them and stay with a `HashMap` in this instance. Or we could stay with a `HashMap` for now and do the benchmarking some time later, if they take effort to run.

Reply:
> Or we could stay with a HashMap for now and do the benchmarking some time later, if they take effort to run.

I decided on this approach, in the interest of getting this merged quickly, to avoid conflicts.

Reply:
I ran the queries used previously for top-k evaluation with the `BTreeMap` change. The results are documented in PR #17401. Note, however, that the queries there are in a setup with only a single group to stress worst-case skew in monotonic top-k computation. If we feel that we need to extend this, just let me know.